deepdive1/README.md

```diff
@@ -0,0 +1,371 @@
+        MySecondTestClass.class})
+public class UnitTestThatRunsTwoTestClasses {}
+```
+* `@Rule`: [How do rules work?](http://stackoverflow.com/questions/13489388/how-does-junit-rule-work) & [Rules in Android.](http://wiebe-elsinga.com/blog/whats-new-in-android-testing/)
+
+### Important resources linked above, for your convenience
+* [TDD](https://www.agilealliance.org/glossary/tdd/) and [Continuous Integration](https://www.thoughtworks.com/continuous-integration)
+* Read through [this](https://developer.android.com/training/testing/unit-testing/index.html) if you're interested in learning more about testing. This README.md is heavily based off of this.
+* [Local unit tests](https://developer.android.com/training/testing/unit-testing/local-unit-tests.html#build)
+* [Mockito](http://mockito.org/)
+* [Assert](http://junit.sourceforge.net/javadoc/org/junit/Assert.html)
+* [Hamcrest matchers](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html) (and a [quick intro](http://www.slideshare.net/shaiyallin/hamcrest-matchers) too)
+* [Annotations](http://junit.sourceforge.net/javadoc/org/junit/package-summary.html)
+* [Instrumented tests with Espresso](https://google.github.io/android-testing-support-library/docs/espresso/basics/index.html)
+* [ViewMatchers docs](https://developer.android.com/reference/android/support/test/espresso/matcher/ViewMatchers.html)
+    * [ViewMatchers source code](https://android.googlesource.com/platform/frameworks/testing/+/android-support-test/espresso/core/src/main/java/android/support/test/espresso/matcher/ViewMatchers.java) 
+* [ViewActions docs](https://developer.android.com/reference/android/support/test/espresso/action/ViewActions.html)
+    * [ViewActions source code](https://android.googlesource.com/platform/frameworks/testing/+/android-support-test/espresso/core/src/main/java/android/support/test/espresso/action/ViewActions.java)
+* [Make your own matcher (`BoundedMatcher`)](http://www.programcreek.com/java-api-examples/index.php?api=android.support.test.espresso.matcher.BoundedMatcher)
+
+### Extra resources
+* [Presentation](https://docs.google.com/presentation/d/18Gb8NIR_e-cB_LK4evxB7SrdPyrmlxG0GKL5-qcprnA/edit?usp=sharing) from in-class.
+* [Monkey](https://developer.android.com/studio/test/monkey.html) and [MonkeyRunner](https://developer.android.com/studio/test/monkeyrunner/index.html) for how to do stress tests (making your tests even more comprehensive).
+* [Examples of Espresso tests](https://google.github.io/android-testing-support-library/samples/)
+
+## Activity
+As an in-class activity, **write 3 tests** for the "Intro to Java" lesson. Then, try **writing 3 tests** for any of the Android apps. Dividing tests between assignments is fine. Note that this is a non-graded in-class activity, not homework, so we aren't expecting you to finish and turn it in.
+
+### Check your understanding
+We hope that at the end of the lesson, you can answer all these questions:
```

Comment:
oh woops I meant "compare their knowledge [if I ask them to raise their hands]". i'm tired.

anyway, yeah I'll have them talk to the person next to them and talk about those questions together :+1: